### PR TITLE
Fix the bug of 'exclude mode' and boolean values

### PR DESCRIPTION
**Pull request by chris110408**
Fix 'exclude mode' bug for facet type
Fix boolean values zero count search bug for facet type

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Your PR has lint warnings. Please consider fixing these.
      
<details>
  <summary>facet.js</summary>
  <p><a href="https://github.com/smartprocure/contexture-mongo/blob/f/exclude/s" target="_blank">s</a></p>
  
  <strong>'agg' is defined but never used.</strong>
  <code>Line 552: agg => [],</code>
</details>
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 27f7475bb96d4ebc71ecfe699a0cb0bcf01172dc


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f0a451956c5272f0848a2364e0dd80835d5bea75
</p>

File: test/example-types/facet.js [hunk 1]
```diff
@@ -507,27 +520,61 @@ describe('facet', () => {
...
+          }),
+          getSchema() {},
+        }
+        node.size = 1
+        node.field = 'bool'
+        ;(node.label = {
+          collection: mongoIdData,
+          foreignField: '_id',
+        }),
+          (node.values = [true])
```
Comment (daedalus28) [inline test/example-types/facet.js:572]: what is this weird syntax? 🤔

File: src/example-types/facet.js [hunk 2]
```diff
@@ -175,11 +176,13 @@ module.exports = {
           mapKeywordFilters(node),
         ])
       )
+
       let zeroCountIds = _.difference(
         //when values are numeric values, stringify missedValues to avoid the bug.
-        _.map(_.toString, lostIds),
-        _.map(x => _.toString(x[`${node.field}`]), lostOptions)
+        _.map(F.unless(_.isBoolean, _.toString), lostIds),
+        _.map(({ _id }) => _.toString(_id), lostOptions)
```
Comment (daedalus28) [inline src/example-types/facet.js:183]: are you sure this should be `_id` and not the node field?

**Review by daedalus28 — CHANGES_REQUESTED** [review]

File: src/example-types/facet.js (hunk 2 above)
Comment (chris110408) [reply src/example-types/facet.js:183]: Yes, I am sure. This is really hard to find a bug, https://github.com/smartprocure/contexture-mongo/blob/5e1e876bd8d0a42f3ab8fbaa1bf0ecfafa52a76f/src/example-types/facet.js#L172 will convert the field name to id

File: test/example-types/facet.js (hunk 1 above)
Comment (chris110408) [reply test/example-types/facet.js]: Duti changed my code to this weird format. cause I had an extra comma

**Review by daedalus28 — APPROVED** [review]

**Pull request merged**